### PR TITLE
Make hardware acceleration configurable.

### LINKED ISSUE #75: Require hardware acceleration
something like 

```decoder.configure({ codec: 'vp8', acceleration: 'require' })```

which should fail to configure if hw acceleration is not available for the given configuration. Same thing encoder. 

This allows apps who bring their own software codecs in WASM to prefer those when we would otherwise be using software internally. 

We have [a version of this enum](https://source.chromium.org/chromium/chromium/src/+/master:third_party/blink/renderer/modules/webcodecs/video_encoder_config.idl;drc=e9b426c19ec3dfc778eea5da7e040f90f8260377;l=9) already in Chromium for the encoder. We should make it a more generically named thing to share w/ decode and formally spec it.

### PR DESCRIPTION
**Pull request by chcunningham**
Fixes #75.

***
<a href="https://pr-preview.s3.amazonaws.com/WICG/web-codecs/pull/118.html" title="Last updated on Feb 23, 2021, 12:33 AM UTC (3f5f2ed)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/WICG/web-codecs/118/f625d42...3f5f2ed.html" title="Last updated on Feb 23, 2021, 12:33 AM UTC (3f5f2ed)">Diff</a>

### CODE REVIEW

**Review by padenot — COMMENTED** [review]
In general, looks good. Maybe adding an informative note to  explain the usefulness of this (it's not obvious), and I'm not sure about the name `acceleration`, it doesn't mention "hardware" acceleration. Maybe simply `hardwareAcceleration` ?

File: index.src.html [hunk 1]
```diff
@@ -1295,6 +1303,12 @@
 <dl>
   <dt><dfn dict-member for=VideoEncoderConfig>avc</dfn></dt>
   <dd>Contains codec specific configuration options for the AVC (H.264) codec.</dd>
+
+  <dt><dfn dict-member for=VideoEncoderConfig>acceleration</dfn></dt>
+  <dd>
+    Configures hardware accleration for this codec. See
```
[the remark this answers is not in the document]
Comment (chcunningham) [reply index.src.html]: fixed

**Review by chcunningham — COMMENTED** [review]
> Maybe adding an informative note to explain the usefulness of this (it's not obvious)

Done

> and I'm not sure about the name acceleration, it doesn't mention "hardware" acceleration. Maybe simply hardwareAcceleration ?

Done

**Comment by chcunningham** [comment]
@padenot I'll go ahead with this merge and address any follow up feedback when you return.

**Pull request merged**